staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -405,7 +405,7 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 // a node object with provider-specific information.
 // All of the returned functions are idempotent, because they are used in a retry-if-conflict
 // loop, meaning they could get called multiple times.
-func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Context, node *v1.Node) ([]nodeModifier, error) {
```

Comment:
I don't think it makes sense for this method to be returning `InstanceMetadata`, what was the reason for returning it here?

Reply:
Returning `InstanceMetadata` and delivering it to `updateNodeAddress`, then `updateNodeAddress` won't call cloud apis to get `InstanceMetadata` again.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L393

Reply:
Could we get `InstanceMetadata` in a separate method before and pass it into `getNodeModifiersFromCloudProvider` instead?

Reply:
Sure we can, but we should move get providerID outside `getNodeModifiersFromCloudProvider` at the same time.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L415-L436

Is it ok?

Reply:
I think so

Reply:
Done, PTAL.